### ISSUE TITLE
Webgl example1 (and 2) doesn't work on windows because of "symbolic link"

### ISSUE THREAD

**Issue by stonebig** [vispy/vispy#907]
hi,

On today master, I try the two webgl examples in Ipython3.1/windows7/python3.4.3, and I fall on this error:

for the first cell of webgl_example_1
````
import numpy as np
import vispy
import vispy.gloo as gloo
from vispy import app
from vispy.util.transforms import perspective, translate, rotate
app.use_app('ipynb_webgl');
````

````
symlink D:\result_tests\winpython-3.4.3.3\settings\.ipython\nbextensions\vispy -> D:\result_tests\winpython-3.4.3.3\python-3.4.3\lib\site-packages\vispy\html\static\js

:0: FutureWarning: IPython widgets are experimental and may change in the future.

---------------------------------------------------------------------------
OSError                                   Traceback (most recent call last)
<ipython-input-1-178aece809a4> in <module>()
      4 from vispy import app
      5 from vispy.util.transforms import perspective, translate, rotate
----> 6 app.use_app('ipynb_webgl');

D:\result_tests\winpython-3.4.3.3\python-3.4.3\lib\site-packages\vispy\app\_default_app.py in use_app(backend_name, call_reuse)
     44 
     45     # Create default app
---> 46     default_app = Application(backend_name)
     47     return default_app
     48 

D:\result_tests\winpython-3.4.3.3\python-3.4.3\lib\site-packages\vispy\app\application.py in __init__(self, backend_name)
     47         self._backend_module = None
     48         self._backend = None
---> 49         self._use(backend_name)
     50 
     51     def __repr__(self):

D:\result_tests\winpython-3.4.3.3\python-3.4.3\lib\site-packages\vispy\app\application.py in _use(self, backend_name)
    225 
    226         # Store classes for app backend and canvas backend
--> 227         self._backend = self.backend_module.ApplicationBackend()

D:\result_tests\winpython-3.4.3.3\python-3.4.3\lib\site-packages\vispy\app\backends\_ipynb_webgl.py in __init__(self)
     69     def __init__(self):
     70         BaseApplicationBackend.__init__(self)
---> 71         _prepare_js()
     72 
     73     def _vispy_reuse(self):

D:\result_tests\winpython-3.4.3.3\python-3.4.3\lib\site-packages\vispy\app\backends\_ipynb_webgl.py in _prepare_js(force)
     62         kwargs = {}
     63     install_nbextension(jsdir, overwrite=force, destination='vispy',
---> 64                         symlink=True, **kwargs)
     65 
     66 

D:\result_tests\winpython-3.4.3.3\python-3.4.3\lib\site-packages\IPython\html\nbextensions.py in install_nbextension(path, overwrite, symlink, user, prefix, nbextensions_dir, destination, verbose)
    225                 if verbose >= 1:
    226                     print("symlink %s -> %s" % (full_dest, path))
--> 227                 os.symlink(path, full_dest)
    228         elif os.path.isdir(path):
    229             path = pjoin(os.path.abspath(path), '') # end in path separator

OSError: symbolic link privilege not held

````

I guess it's a windows limitation. Is there any workaround ?

**Comment by rossant** [comment]
I guess this is an ipython issue? You may want to ask them...

**Comment by stonebig** [comment]
ah, If it's not a Windows issue, hope is back.... digging further

**Comment by stonebig** [comment]
Well, it seems it means I'm not administrator of my machine, and users by default can't create symlinks

**Comment by Eric89GXL** [comment]
Yeah, you'll need an admin to tweak a setting:

http://superuser.com/questions/124679/how-do-i-create-a-link-in-windows-7-home-premium-as-a-regular-user

**Comment by stonebig** [comment]
I didn't have this Windows problem before.
As a workaround, Will this work ?

replacing
````
mklink /D "D:\result_tests\winpython-3.4.3.3\python-3.4.3\lib\site-packages\vispy\html\static\js" "D:\result_tests\winpython-3.4.3.3\settings\.ipython\nbextensions\vispy"
````
per 
````
copy "D:\result_tests\winpython-3.4.3.3\settings\.ipython\nbextensions\vispy" "D:\result_tests\winpython-3.4.3.3\python-3.4.3\lib\site-packages\vispy\html\static\js" 
````

**Comment by Eric89GXL** [comment]
I'm not sure, but I suspect that a copy will work instead of a symlink, yeah

**Comment by stonebig** [comment]
ok, so I intercept the error in the Ipython code, to see further

**Comment by stonebig** [comment]
ok, after the manual copy, Ipython decides to avoid the symlink, but then nothing shows up.

**Comment by stonebig** [comment]
heum, a little better with Chrome instead of firefox

**Comment by stonebig** [comment]
ok, works on chrome. my firefox is mayby disabling too much scripts.
So the issue is that, on windows, the symlink would be better replace per a copy, or another solution

**Comment by stonebig** [comment]
Webgl example2 is puting chrome on his knees, I don't guess exactly why, as memory and Cpu usage was looking rather constant.

**Comment by rossant** [comment]
you have quite some settings to do on chrome to enable gpu acceleration in
webgl
you'll find some references on google

2015-05-16 19:34 GMT+02:00 stonebig <notifications@github.com>:

**Comment by stonebig** [comment]
yes:
````
chrome://gpu/
````
it says that my i3-350m is `WebGL: Hardware accelerated`, but that may not mean enough for your example.

**Comment by stonebig** [comment]
Why did yo pass the parameter `symlink=True` in `vispy/app/backends/_ipynb_webgl.py` ?

Could you switch it to `False` under windows ?

**Comment by rossant** [comment]
that's better for development, but i guess we could put it to false for end-users...
feel free to test it and propose a PR if it works!

**Comment by stonebig** [comment]
ok, there #909 .

Next it the curious behaviour:
- first time I run webgl_example_1, I get a dark square,
- then if I destroy the black square and re-rerun from first cell, it becomes OK.

==> Could there be a wrong ordering of orders, that get fixed by running twice ?

**Comment by rossant** [comment]
any JS console output (F12 dev tools)?

cc @cgestes

2015-05-17 12:37 GMT+02:00 stonebig <notifications@github.com>:

**Comment by stonebig** [comment]
this ?
![vispy_webgl1_js_console](https://cloud.githubusercontent.com/assets/4312421/7670444/d3e35e1c-fca4-11e4-95fe-9235c0119459.GIF)

**Comment by stonebig** [comment]
I don't guess on retry what is looking different, but it works:
![vispy_webgl1_js_console_on_retry_ok](https://cloud.githubusercontent.com/assets/4312421/7670457/57e11146-fca5-11e4-8d5b-ac8ad8dd9348.GIF)

**Comment by rossant** [comment]
I can't reproduce this on Linux
might have to do with async loading
@cgestes have you seen this already?

**Comment by Eric89GXL** [comment]
I think this was fixed by #909 feel free to reopen if I'm mistaken
